### PR TITLE
Allow custom args to be passed through to main class in run-class.sh.

### PR DESCRIPTION
**Pull request by danxmoran**
Required for specifying custom solver ports in aristo

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 target/
+.idea/
```
Comment (markschaake) [inline .gitignore]: You should set up global excludes on your machine and remove this IDE-specific ignore from the PR. See http://stackoverflow.com/questions/7335420/global-git-ignore

Comment (danxmoran) [inline .gitignore]: Thanks!

File: src/main/resources/run-class.sh [hunk 2]
```diff
@@ -19,12 +22,12 @@ fi
 SCRIPT_NAME=`basename $0`
 # Use a compact usage if we're being invoked as another script.
 if [[ $SCRIPT_NAME == run-class.sh ]]; then
-  USAGE="usage: $SCRIPT_NAME mainClass shortName [start|stop|restart]"
+  USAGE="usage: $SCRIPT_NAME mainClass shortName [start|stop|restart] [ -- args* ]"
 else
-  USAGE="usage: $SCRIPT_NAME [start|stop|restart]"
+  USAGE="usage: $SCRIPT_NAME [start|stop|restart] [ -- args* ]"
```
Comment (jkinkead) [inline src/main/resources/run-class.sh]: More standard would be `[-- <args>]`.

File: src/main/resources/run-class.sh [hunk 3]
```diff
@@ -51,6 +54,14 @@ stop)
   exit 1
 esac
 
+if [[ ! -z "$1" ]]; then
```
Comment (jkinkead) [inline src/main/resources/run-class.sh]: Comment?

File: src/main/resources/run-class.sh [hunk 4]
```diff
@@ -111,8 +122,8 @@ JAVA_CMD=(java $JVM_ARGS -classpath $CLASSPATH $CONF_FILE
 
 # Run java.
 echo "running in `pwd` ..."
-echo "${JAVA_CMD[@]} ${MAIN_CLASS}"
-nohup ${JAVA_CMD[@]} "$MAIN_CLASS" > "${SHORT_NAME}.out" 2> "${SHORT_NAME}.err" &
+echo "${JAVA_CMD[@]} ${MAIN_CLASS} ${@}"
```
Comment (jkinkead) [inline src/main/resources/run-class.sh]: Remove braces around `@` here?

**Comment by jkinkead** [comment]
A couple of small comments.

**Comment by danxmoran** [comment]
Comments addressed!

File: src/main/resources/run-class.sh [hunk 5]
```diff
@@ -51,6 +54,15 @@ stop)
   exit 1
 esac
 
+# Check for any additional args to pass through to the main java class, separated by "--".
```
Comment (jkinkead) [inline src/main/resources/run-class.sh]: You're really checking for a `--` and stripping it, giving an error if it's missing.

**Comment by jkinkead** [comment]
Comment nitpick. Feel free to merge when you're happy!

**Pull request merged**